app/assets/javascripts/discourse/app/services/document-title.js

```diff
@@ -0,0 +1,105 @@
+import Service from "@ember/service";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+
+export default Service.extend({
+  appEvents: service(),
+  contextCount: null,
+  notificationCount: null,
+  _title: null,
+  _backgroundNotify: null,
+
+  init() {
+    this._super(...arguments);
+    this.reset();
+  },
+
+  reset() {
+    this.contextCount = 0;
+    this.notificationCount = 0;
+    this._title = null;
+    this._backgroundNotify = null;
+  },
+
+  getTitle() {
+    return this._title;
+  },
+
+  setTitle(title) {
+    this._title = title;
+    this._renderTitle();
+  },
+
+  updateFocus(focus) {
```

Comment:
That's a nitpick here but could me try to have the same naming pattern:

setTitle
setFocus
setContextCount
setNotificationCount

(or update instead of set)